flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/BenchmarkResult.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.benchmark;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/** The result of executing a benchmark. */
+public class BenchmarkResult {
+    /** The name of the benchmark. */
+    public String name;
```

Comment:
I agree that these fields should be made `final`, but adding a constructor with all these field values might possibly limit the extension of the interface. For example, when in near future when we add fields like `latencyP99Ms`, all usages of this constructor needs to be modified. I would prefer to add a `BenchmarkResult.Builder` that helps setting an arbitrary number of fields before constructing the `BenchmarkResult` instance.

Reply:
hmm... in the future when we add `latencyP99Ms`, we can just add a new constructor and update the existing constructors to provide a default value for `latencyP99Ms`. We want need to change all existing usages, right?

Note that if we need to change existing usages, it should be because we want to provide a valid non-default value for `latencyP99Ms`. We will also need to do these changes if we use `Builder` in this case.

The usages of multiple constructors is a common pattern across Flink to handle scenarios like this.